### PR TITLE
[chore] implemented zextension server using httpServerSettings

### PR DESCRIPTION
**Pull request by Sanket-0510**
**Description:** <Describe what has changed.>
implementing z-extension server using `httpServerSetting`  instead of normal `http.server`

**Link to tracking Issue:** #9368 

**Testing:** I performed `make` but am facing some linting issues definitely I need help with this 

Screenshot from 2024-01-25 16-27-19

### CODE REVIEW

File: extension/zpagesextension/zpagesextension.go [hunk 1]
```diff
@@ -73,7 +75,12 @@ func (zpe *zpagesExtension) Start(_ context.Context, host component.Host) error
 	}
 
 	zpe.telemetry.Logger.Info("Starting zPages extension", zap.Any("config", zpe.config))
-	zpe.server = http.Server{Handler: zPagesMux}
+	server, err := zpe.httpSettings.ToServer(host,zpe.telemetry,zPagesMux)
+	if err != nil {
+		return err
+	}
+    zpe.server = server
```
Comment (atoulme) [inline extension/zpagesextension/zpagesextension.go:82]:
```suggestion
	zpe.server = server
```

File: extension/zpagesextension/zpagesextension.go [hunk 2]
```diff
@@ -73,7 +75,12 @@ func (zpe *zpagesExtension) Start(_ context.Context, host component.Host) error
 	}
 
 	zpe.telemetry.Logger.Info("Starting zPages extension", zap.Any("config", zpe.config))
-	zpe.server = http.Server{Handler: zPagesMux}
+	server, err := zpe.httpSettings.ToServer(host,zpe.telemetry,zPagesMux)
```
Comment (atoulme) [inline extension/zpagesextension/zpagesextension.go:78]:
```suggestion
	server, err := zpe.httpSettings.ToServer(host,zpe.telemetry, zPagesMux)
```

File: extension/zpagesextension/zpagesextension.go [hunk 3]
```diff
@@ -24,8 +25,9 @@ type zpagesExtension struct {
 	config              *Config
 	telemetry           component.TelemetrySettings
 	zpagesSpanProcessor *zpages.SpanProcessor
-	server              http.Server
+	server              *http.Server
 	stopCh              chan struct{}
+	httpSettings        confighttp.HTTPServerSettings
```
Comment (atoulme) [inline extension/zpagesextension/zpagesextension.go:30]: that's not quite right, this should be in the Config struct.

Comment (Sanket-0510) [reply extension/zpagesextension/zpagesextension.go:30]: can you elaborate more over this

Comment (atoulme) [reply extension/zpagesextension/zpagesextension.go:30]: Here is an example of use: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/config.go#L28
Here is how it's used https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/healthcheckextension.go#L35-L41

Comment (Sanket-0510) [reply extension/zpagesextension/zpagesextension.go]: I have added it to the Config struct and have taken care for the types of the parameters being passed by the methods but dont know why am failing over this test which is related to difference between the Endpoint found and the given in the test.

Test Failure log :

```
config_test.go:32: 
                Error Trace:    /media/sanket-teli/X1/opentelemetry-collector/extension/zpagesextension/config_test.go:32
                Error:          Not equal: 
                                expected: &zpagesextension.Config{HTTPServerSettings:confighttp.HTTPServerSettings{Endpoint:"", TLSSetting:(*configtls.TLSServerSetting)(nil), CORS:(*confighttp.CORSSettings)(nil), Auth:(*configauth.Authentication)(nil), MaxRequestBodySize:0, IncludeMetadata:false, ResponseHeaders:map[string]configopaque.String(nil)}, TCPAddr:confignet.TCPAddr{Endpoint:"localhost:56888", DialerConfig:confignet.DialerConfig{Timeout:0}}}
                                actual  : &zpagesextension.Config{HTTPServerSettings:confighttp.HTTPServerSettings{Endpoint:"localhost:56888", TLSSetting:(*configtls.TLSServerSetting)(nil), CORS:(*confighttp.CORSSettings)(nil), Auth:(*configauth.Authentication)(nil), MaxRequestBodySize:0, IncludeMetadata:false, ResponseHeaders:map[string]configopaque.String(nil)}, TCPAddr:confignet.TCPAddr{Endpoint:"localhost:56888", DialerConfig:confignet.DialerConfig{Timeout:0}}}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,3 +2,3 @@
                                  HTTPServerSettings: (confighttp.HTTPServerSettings) {
                                -  Endpoint: (string) "",
                                +  Endpoint: (string) (len=15) "localhost:56888",
                                   TLSSetting: (*configtls.TLSServerSetting)(<nil>),
                Test:           TestUnmarshalConfig
2024/01/27 04:48:37 http: panic serving 127.0.0.1:42552: runtime error: invalid memory address or nil pointer dereference

```

**Comment by atoulme** [comment]
Please rebase, you seem to have many conflicts.

File: extension/zpagesextension/zpagesextension.go [hunk 4]
```diff
@@ -73,7 +73,12 @@ func (zpe *zpagesExtension) Start(_ context.Context, host component.Host) error
 	}
 
 	zpe.telemetry.Logger.Info("Starting zPages extension", zap.Any("config", zpe.config))
-	zpe.server = http.Server{Handler: zPagesMux}
+	server, err := zpe.config.ToServer(host, zpe.telemetry, nil)
```
Comment (atoulme) [inline extension/zpagesextension/zpagesextension.go:76]: Do you not want to pass the mux that serves http?
```suggestion
	server, err := zpe.config.ToServer(host, zpe.telemetry, zPagesMux)
```

Comment (Sanket-0510) [reply extension/zpagesextension/zpagesextension.go:76]: Actually I tried this but again the same test is failing, I think I will have to look at the implementation carefully or may be there are some more things I would need to explore in order to address  the Z-page extension server using configHttp module

**Comment by Sanket-0510** [comment]
Hey @atoulme I need help with this one. config_test is failing, can you elaborate what else needs to be done in this.

File: extension/zpagesextension/config.go [hunk 5]
```diff
@@ -7,11 +7,13 @@ import (
 	"errors"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/config/confignet"
 )
 
 // Config has the configuration for the extension enabling the zPages extension.
 type Config struct {
+	confighttp.ServerConfig `mapstructure:",squash"`
```
Comment (astencel-sumo) [inline extension/zpagesextension/config.go:16]: You are adding `confighttp.ServerConfig`, which includes the `Endpoint` property. You should remove the `TCPAddr` then, as it is not needed anymore.

**Pull request closed without merging**